name: .NET Core

on: 
  push:
    branches: manohar-docker
    paths: [apis/poi/**,.github/workflows/workflowarchna.yml]
  pull_request:
    branches: master
    paths: [apis/poi/**,.github/workflows/workflowarchna.yml]

jobs:
  build:
    runs-on: ubuntu-latest
    #continue-on-error: true
    steps:
    - uses: actions/checkout@v2
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 3.1.301
    - name: Install dependencies
      run: dotnet restore apis/poi/web/poi.csproj
    - name: Build
      run: dotnet build --configuration Release --no-restore apis/poi/web/poi.csproj
    - name: Test      
      run: dotnet test --no-restore --verbosity normal apis/poi/tests/UnitTests/UnitTests.csproj 
    - name: Open Github Issue
      if: ${{ failure() }}
      uses: rishabhgupta/git-action-issue@v2
      id: Issue
      with:
          token: ${{ secrets.GITHUB_TOKEN }}
          title: Automated Issue creation
          body: This issue is created because tests have failed
          assignees: 'archnaaju'
    - name: Build image
      run: |
        docker build -t openhackxfa5tiw4acr.azurecr.io/${{ secrets.ACR_USER }}/poi:${{github.run_number}} --build-arg SQL_USER=${{secrets.DB_USER_NAME}} --build-arg SQL_PASSWORD=${{secrets.DB_PASSWORD}} --build-arg SQL_SERVER=${{secrets.DB_SERVER}} --build-arg SQL_DBNAME=${{secrets.DB_NAME}} --build-arg WEB_SERVER_BASE_URI=https://openhackxfa5tiw4poi.azurewebsites.net apis/poi/web
    - name: acr login
      run: docker login openhackxfa5tiw4acr.azurecr.io -u ${{ secrets.ACR_USER }} -p ${{ secrets.ACR_KEY }}
    - name: docker push
      run: docker push openhackxfa5tiw4acr.azurecr.io/${{ secrets.ACR_USER }}/poi:${{github.run_number}} 
        
      
